ygen/yang_helpers.go

```diff
@@ -408,3 +410,30 @@ func isYANGBaseType(t *yang.YangType) bool {
 	_, builtin := yang.TypeKindFromName[t.Name]
 	return builtin
 }
+
+// typeDefaultValue returns the default value of the type t if it is specified.
+// nil is returned if no default is specified.
+func typeDefaultValue(t *yang.YangType) *string {
+	if t.Default == "" {
+		return nil
+	}
+	return ygot.String(t.Default)
+}
+
+// enumDefaultValue sanitises a default value specified for an enumeration
+// which can be specified as prefix:value in the YANG schema. The baseName
+// is used as the generated enumeration name stripping any prefix specified,
+// (allowing removal of the enumeration type prefix if required). The default
+// value in the form <sanitised_baseName>_<sanitised_defVal> is returned as
+// a pointer.
+func enumDefaultValue(baseName, defVal, prefix string) *string {
+	if strings.Contains(defVal, ":") {
+		defVal = strings.SplitN(defVal, ":", -1)[1]
```

Comment:
(optional) If we are passing -1 as the split count, we could as well use strings.Split() instead of strings.SplitN().

Reply:
Done.